test/functional/action-params.spec.ts

```diff
@@ -72,6 +78,46 @@ describe("action parameters", () => {
         // reset metadata args storage
         defaultMetadataArgsStorage().reset();
 
+
+        @Middleware()
+        class SessionMiddleware implements MiddlewareInterface {
+            public use (requestOrContext: any, responseOrNext: any, next?: (err?: any) => any): any {
+                if (next) {
+                    return this.expSession(requestOrContext, responseOrNext, next);
+                } else {
+                    if (!this.koaSession) {
+                        this.koaSession = convert(KoaSession(requestOrContext.app));
+                    }
+                    return this.koaSession(requestOrContext, responseOrNext);
+                }
+            }
+
+            private expSession = session({
+                secret: "19majkel94_helps_pleerock",
+            });
+
+            private koaSession: any;
+        }
+
+        class User {
+            public username: string;
+            public location: string;
+            public twitter: string;
+        }
+
+        @Middleware()
+        class SetStateMiddleware implements MiddlewareInterface {
```

Comment:
`SetStateMiddleware` should be moved to `test/fakes/global-options/koa-middlewares`